docs/docs-content/clusters/public-cloud/azure/required-permissions.md

```diff
@@ -46,254 +46,893 @@ You can use Validator with the Palette CLI. Check out the Palette CLI
+            "Microsoft.Compute/virtualMachines/extensions/write",
+            "Microsoft.Compute/virtualMachines/read",
+            "Microsoft.Compute/virtualMachines/write",
+            "Microsoft.Network/loadBalancers/backendAddressPools/join/action",
+            "Microsoft.Network/loadBalancers/delete",
+            "Microsoft.Network/loadBalancers/inboundNatRules/delete",
+            "Microsoft.Network/loadBalancers/inboundNatRules/join/action",
+            "Microsoft.Network/loadBalancers/inboundNatRules/read",
+            "Microsoft.Network/loadBalancers/inboundNatRules/write",
+            "Microsoft.Network/loadBalancers/read",
+            "Microsoft.Network/loadBalancers/write",
+            "Microsoft.Network/networkInterfaces/delete",
+            "Microsoft.Network/networkInterfaces/join/action",
+            "Microsoft.Network/networkInterfaces/read",
+            "Microsoft.Network/networkInterfaces/write",
+            "Microsoft.Network/networkSecurityGroups/read",
+            "Microsoft.Network/networkSecurityGroups/securityRules/delete",
+            "Microsoft.Network/networkSecurityGroups/securityRules/read",
+            "Microsoft.Network/networkSecurityGroups/securityRules/write",
+            "Microsoft.Network/privateDnsZones/A/delete",
+            "Microsoft.Network/privateDnsZones/A/read",
+            "Microsoft.Network/privateDnsZones/A/write",
+            "Microsoft.Network/privateDnsZones/delete",
+            "Microsoft.Network/privateDnsZones/read",
+            "Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete",
+            "Microsoft.Network/privateDnsZones/virtualNetworkLinks/read",
+            "Microsoft.Network/privateDnsZones/virtualNetworkLinks/write",
+            "Microsoft.Network/privateDnsZones/write",
+            "Microsoft.Network/publicIPAddresses/delete",
+            "Microsoft.Network/publicIPAddresses/join/action",
+            "Microsoft.Network/publicIPAddresses/read",
+            "Microsoft.Network/publicIPAddresses/write",
+            "Microsoft.Network/routeTables/delete",
+            "Microsoft.Network/routeTables/read",
+            "Microsoft.Network/routeTables/write",
+            "Microsoft.Network/virtualNetworks/join/action",
+            "Microsoft.Resources/subscriptions/resourceGroups/read",
+            "Microsoft.Network/virtualNetworks/read",
+            "Microsoft.Network/virtualNetworks/subnets/join/action",
+            "Microsoft.Network/virtualNetworks/subnets/read",
+            "Microsoft.Compute/galleries/images/read",
+            "Microsoft.Compute/galleries/images/versions/read"
+          ],
+          "AssignableScopes": ["/subscriptions/$SUBSCRIPTION_ID"]
+        }
+        ```
+    </details>
+
+3.  Create a role for the JSON file you created in the previous step. Issue the following command to create the role.
+
+    ```shell
+    az role definition create --role-definition static-placement-permissions.json
```

Comment:
Looks like we might not even need the intermediate file step. Interesting...

![image](https://github.com/user-attachments/assets/e71702ff-73c4-46f3-b60b-d118d47b226d)

I originally proposed to Karl we do it this way with the intermediate file because I only saw it done that way in Azure docs. But if we're only using the file for the step where we create the role definition, there's no need to even have that file. We can just give users `az` commands that inline the JSON.

Reply:
Awesome catch @addetz